packages/keyring-controller/src/KeyringController.ts

```diff
@@ -372,6 +372,16 @@ export class KeyringController extends BaseController<
     return this.#keyring.getAccounts();
   }
 
+  /**
+   * Returns all keyrings of the given type.
+   *
+   * @param type - Keyring type name.
+   * @returns An array of keyrings of the given type.
+   */
+  getKeyringsByType(type: KeyringTypes | string): Keyring[] {
```

Comment:
Here we don't actually return a `Keyring` array, but an array of instances of keyrings. 
@Gudahtt I guess that at some point we might want to use [this one](https://github.com/MetaMask/utils/blob/main/src/keyring.ts#L47) instead of the interface we use as `Keyring` now in KeyringController

Reply:
Interesting, I hadn't realized this `Keyring` state existed.

At some point we should stop returning the keyring state from _any_ function, that's an older pattern that predates our current state stores. Once we have a state store setup (after the v2 migration), returning state from methods will be redundant. That would make it easy to get rid of the `Keyring` type here as well, since it will become redundant with the state type.

Perhaps in the meantime we can make this type less-incorrect at least, e.g. return `unknown` or something.

Reply:
@Gudahtt It does make logical sense that the return type of this function should match the return type of `getKeyringsByType` in `@metamask/eth-keyring-controller`, but is the issue that this would be difficult to do, or is that it's something we are getting rid of soon anyway so it doesn't matter what the return type of this method is?

Reply:
There _is_ no return type for that method right now because the published version of that package is not in TypeScript yet. The next release will include the TypeScript conversion, but it didn't seem necessary for this issue to be blocked by that.